lib/ansible/modules/system/env_file.py

```diff
@@ -0,0 +1,249 @@
+description:
+    - Set permanent environment variables
+    - Currently developed for Debian, Ubuntu, RedHat, CentOS
+    - Makes use of the /etc/environment file
+notes:
+    - Supports check mode
+    - Supports diff
+options:
+    key:
+        required: true
+        aliases: ["name"]
+        description:
+            - unique identifier for environment value (key)
+    value:
+        required: false
+        description:
+            - defines the value for your identifier/key
+    state:
+        type: bool
+        required: true
+        choices: ["present", "absent"]
+        default: present
+        description:
+            - if present add environment variable
+            - if absent delete environment variable
+    force:
+        type: bool
+        required: false
+        default: true
+        description:
+            - if false dont overwrite variables with same key but different content
+'''
+
+EXAMPLES = '''
+- env_file:
+    name: http_proxy
+    value: 127.0.0.1
+    state: present
+
+- env_file:
+    key: ftp_proxy
+    state: absent
+
+'''
+
+RETURN = '''#'''
+
+import os
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import get_exception
+from ansible.module_utils._text import to_text
+from ansible.module_utils._text import to_bytes
+
+
+ENVFILE = "/etc/environment"
+
+
+def get_diff(before, after):
+    before_d = {}
+    for k, v in before.items():
+        before_d[k] = v.replace('"', '').replace("\n", "")
+
+    after_d = {}
+    for k, v in after.items():
+        after_d[k] = v.replace('"', '').replace("\n", "")
+
+    diff = {'before': before_d, 'after': after_d}
+    return diff
+
+
+def read_environment(module):
+    d = {}
+    with open(ENVFILE, 'rb') as f:
+        b_data = f.read().splitlines()
+        for line in b_data:
+            try:
+                line_text = to_text(line)
+            except UnicodeError:
+                module.fail_json(msg="There was an error converting content of {0} as binary to text: {1}".format(ENVFILE, get_exception()))
+            try:
+                (k, v) = line_text.split('=')
+            except (ValueError, UnboundLocalError):
+                if get_exception() == ValueError:
+                    pass
+                else:
+                    module.fail_json(msg="There was an error reading the environment vars in {0}. Is your file corrupted?".format(ENVFILE))
+            d[k] = v
+    return d
+
+
+def delete_unnescessary_newlines(module):
+    with open(ENVFILE, 'rb') as f:
+        b_data = f.readlines()
+        for i, l in enumerate(b_data):
+            try:
+                j = to_text(l)
+            except UnicodeError:
+                module.fail_json(msg="There was an error converting content of {0} as binary to text: {1}".format(ENVFILE, get_exception()))
+            if j == "\n":
+                b_data[i] = to_bytes("")
+        with open(ENVFILE, 'wb') as f:
```

Comment:
this is unsafe and can lead to race conditions, you should write to a temp file and use `basic`'s `atomic_move` method  to finalize changes.

Reply:
Thanks! Really appreciate this feedback